lib/beyag/client.rb

```diff
@@ -50,7 +50,11 @@ def connection
         c.options[:timeout] ||= DEFAULT_TIMEOUT
         c.options[:proxy] = proxy if proxy
         c.request :json
-        c.headers = {'Content-Type' => 'application/json'}
+
+        headers = {'Content-Type' => 'application/json'}
+        headers.update(opts[:headers]) if opts[:headers]
+        c.headers = headers
```

Comment:
You can use `c.headers = {'Content-Type' => 'application/json'}.update(opts[:headers].to_h)`